subprojects/core/src/main/java/org/gradle/groovy/scripts/DefaultScriptCompilerFactory.java

```diff
@@ -45,8 +45,8 @@ public ScriptCompilerImpl(ScriptSource source) {
         }
 
         @Override
-        public <T extends Script, M> ScriptRunner<T, M> compile(Class<T> scriptType, CompileOperation<M> extractingTransformer, ClassLoaderScope targetScope, Action<? super ClassNode> verifier) {
-            CompiledScript<T, M> compiledScript = scriptClassCompiler.compile(source, targetScope, extractingTransformer, scriptType, verifier);
+        public <T extends Script, M> ScriptRunner<T, M> compile(Object target, Class<T> scriptType, CompileOperation<M> extractingTransformer, ClassLoaderScope targetScope, Action<? super ClassNode> verifier) {
+            CompiledScript<T, M> compiledScript = scriptClassCompiler.compile(target, source, targetScope, extractingTransformer, scriptType, verifier);
```

Comment:
💅 We should keep the order of parameters meaningful here. I think it makes sense to have parameters related to the source at the beginning, and ones related to target after them. WDYT?